### PR TITLE
[WIP] New additions for v3.4

### PR DESCRIPTION
**Pull request by syurkevi**

### CODE REVIEW

File: arrayfire/algorithm.py [hunk 1]
```diff
@@ -299,6 +299,63 @@ def accum(a, dim=0):
...
+    Returns
+    ---------
+    out : af.Array
+        - will contain scan of input.
+    """
+    out = Array()
+    safe_call(backend.get().af_scan(ct.pointer(out.arr), a.arr, dim, op, inclusive_scan))
+    return out
+
+def scanByKey(key, a, dim=0, op=BINARYOP.BINARY_ADD, inclusive_scan=True):
```
Comment (pavanky) [inline arrayfire/algorithm.py]: Use underscores instead of camel case.

File: arrayfire/algorithm.py [hunk 2]
```diff
@@ -299,6 +299,63 @@ def accum(a, dim=0):
...
+    inclusive_scan: optional: bool. default: True
+        Specifies if the scan is inclusive
+
+    Returns
+    ---------
+    out : af.Array
+        - will contain scan of input.
+    """
+    out = Array()
+    safe_call(backend.get().af_scan(ct.pointer(out.arr), key.arr, a.arr, dim, op, inclusive_scan))
```
Comment (pavanky) [inline arrayfire/algorithm.py]: This should be af_scan_by_key

**Comment by 9prady9** [comment]
@syurkevi are you still working on this ?

**Comment by pavanky** [comment]
@9prady9 I am working on it.

**Comment by 9prady9** [comment]
Can you come online for a chat ?

**Pull request closed without merging**